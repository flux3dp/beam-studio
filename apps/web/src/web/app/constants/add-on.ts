import { WorkAreaModel } from './workarea-constants';

export enum RotaryType {
  Roller = 1,
  Chuck = 2,
}

export const CHUCK_ROTARY_DIAMETER = 133;

export interface SupportInfo {
  autoFocus: boolean;
  hybridLaser: boolean;
  openBottom: boolean;
  passThrough: boolean;
  rotary?: {
    roller: boolean;
    chuck: boolean;
    extendWorkarea: boolean;
    mirror: boolean;
    defaultMirror?: boolean;
  };
  lowerFocus: boolean;
  framingLowLaser?: boolean;
  redLight?: boolean;
  jobOrigin?: boolean;
}

const supportList: Record<WorkAreaModel, SupportInfo> = {
  fbm1: {
    autoFocus: true,
    hybridLaser: true,
    openBottom: true,
    passThrough: true,
    rotary: {
      roller: true,
      chuck: true,
      extendWorkarea: false,
      mirror: false,
    },
    lowerFocus: false,
    jobOrigin: true,
  },
  fbb1b: {
    autoFocus: false,
    hybridLaser: false,
    openBottom: false,
    passThrough: false,
    rotary: {
      roller: true,
      chuck: false,
      extendWorkarea: false,
      mirror: false,
    },
    lowerFocus: false,
    jobOrigin: true,
  },
  fbb1p: {
    autoFocus: false,
    hybridLaser: false,
    openBottom: false,
    passThrough: false,
    rotary: {
      roller: true,
      chuck: false,
      extendWorkarea: false,
      mirror: false,
    },
    lowerFocus: false,
    jobOrigin: true,
  },
  fhexa1: {
    autoFocus: false,
    hybridLaser: false,
    openBottom: false,
    passThrough: false,
    rotary: {
      roller: true,
      chuck: true,
      extendWorkarea: false,
      mirror: false,
    },
    lowerFocus: true,
    jobOrigin: true,
  },
  ado1: {
    autoFocus: false,
    hybridLaser: false,
    openBottom: false,
    passThrough: true,
    rotary: {
      roller: true,
      chuck: true,
      extendWorkarea: true,
      mirror: true,
      defaultMirror: true,
    },
    lowerFocus: true,
    framingLowLaser: true,
    jobOrigin: true,
  },
  fbb2: {
    autoFocus: false,
    hybridLaser: false,
    openBottom: false,
    passThrough: true,
    rotary: {
      roller: true,
      chuck: true,
      extendWorkarea: true,
      mirror: true,
      defaultMirror: true,
    },
    lowerFocus: true,
    redLight: true,
    jobOrigin: true,
  },
  fpm1: {
    autoFocus: false,
    hybridLaser: false,
    openBottom: false,
    passThrough: false,
    rotary: {
      roller: true,
      chuck: false,
      extendWorkarea: false,
      mirror: false,
    },
    lowerFocus: true,
  },
  flv1: {
    autoFocus: false,
    hybridLaser: false,
    openBottom: false,
    passThrough: false,
    rotary: {
      roller: true,
      chuck: false,
      extendWorkarea: false,
      mirror: false,
    },
    lowerFocus: false,
    jobOrigin: true,
  },
};

export const getSupportInfo = (workarea: WorkAreaModel): SupportInfo =>
  supportList[workarea] || {
    autoFocus: false,
    hybridLaser: false,
    openBottom: false,
    passThrough: false,
    lowerFocus: false,
  };

export default {
  getSupportInfo,
  supportList,
};
