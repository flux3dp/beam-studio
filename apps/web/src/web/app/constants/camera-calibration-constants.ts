export const STEP_ASK_READJUST = Symbol('STEP_ASK_READJUST');
export const STEP_REFOCUS = Symbol('STEP_REFOCUS');
export const STEP_PUT_PAPER = Symbol('STEP_PUT_PAPER');
export const STEP_BEFORE_ANALYZE_PICTURE = Symbol('STEP_BEFORE_ANALYZE_PICTURE');
export const STEP_FINISH = Symbol('STEP_FINISH');

export const DEFAULT_CAMERA_OFFSET = {
  X: 15, Y: 30, R: 0, SX: 1.625, SY: 1.625,
};

export const CALIBRATION_PARAMS = {
  centerX: 90, // mm
  centerY: 90, // mm
  size: 25, // mm
  idealOffsetX: 20, // mm
  idealOffsetY: 30, // mm
  idealScaleRatio: (585 / 720) * 2, // pixel on studio / pixel on beambox machine; 與焦距成正比
};
