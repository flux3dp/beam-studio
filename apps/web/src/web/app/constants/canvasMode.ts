export const enum CanvasMode {
  Draw = 1,
  Preview = 2,
  PathPreview = 3,
  CurveEngraving = 4,
}

export default CanvasMode;
