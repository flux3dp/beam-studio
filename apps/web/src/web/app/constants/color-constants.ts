export interface ColorConfig {
  color: string,
  power: number,
  speed: number,
  repeat: number,
}
export const DefaultColorConfigs: ColorConfig[] = [
  {
    color: '#FF0000', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#FFFF00', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#00FF00', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#00FFFF', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#0000FF', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#FF00FF', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#800000', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#808000', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#008000', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#008080', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#000080', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#800080', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#CCCCCC', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#808080', power: 15, speed: 50, repeat: 1,
  },
  {
    color: '#000000', power: 15, speed: 50, repeat: 1,
  },
];

const randomLayerColors = [
  '#333333',
  '#3F51B5',
  '#F44336',
  '#FFC107',
  '#8BC34A',
  '#2196F3',
  '#009688',
  '#FF9800',
  '#CDDC39',
  '#00BCD4',
  '#FFEB3B',
  '#E91E63',
  '#673AB7',
  '#03A9F4',
  '#9C27B0',
  '#607D8B',
  '#9E9E9E',
];

export const objectsColorPresets =[
  '#F5222D',
  '#FA8C16',
  '#FADB14',
  '#8BBB11',
  '#52C41A',
  '#13A8A8',
  '#1677FF',
  '#2F54EB',
  '#722ED1',
];

export enum PrintingColors {
  CYAN = '#009FE3',
  MAGENTA = '#E6007E',
  YELLOW = '#FFED00',
  BLACK = '#1D1D1B',
  WHITE = '#E2E2E2',
}

const printingLayerColor = [
  PrintingColors.CYAN,
  PrintingColors.MAGENTA,
  PrintingColors.YELLOW,
  PrintingColors.BLACK,
  PrintingColors.WHITE,
];

export const CMYK = [
  PrintingColors.CYAN,
  PrintingColors.MAGENTA,
  PrintingColors.YELLOW,
  PrintingColors.BLACK,
];

export default {
  randomLayerColors,
  printingLayerColor,
};
