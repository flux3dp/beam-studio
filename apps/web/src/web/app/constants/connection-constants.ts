// TODO: List all possible reason for
export enum ConnectionError {
  TIMEOUT = 'TIMEOUT',
  UNKNOWN_DEVICE = 'UNKNOWN_DEVICE',
  NOT_FOUND = 'NOT_FOUND',
  DISCONNECTED = 'DISCONNECTED',
  AUTH_ERROR = 'AUTH_ERROR',
  AUTH_FAILED = 'AUTH_FAILED',
  UPDATE_SERIAL_FAILED = 'UPDATE_SERIAL_FAILED',
}

export interface SelectionResult {
  success: boolean;
  error?: ConnectionError;
}
