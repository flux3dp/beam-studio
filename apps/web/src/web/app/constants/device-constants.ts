export default {
  // Status
  READY: 'READY',
  OK: 'OK',
  // folder
  NOT_EXIST: 'NOT_EXIST',
  status: {
    CARTDRIDGE_IO: -17,
    RAW: -10,
    SCAN: -2,
    MAINTAIN: -1,
    IDLE: 0,
    INIT: 1,
    STARTING: 4,
    RESUME_TO_STARTING: 6,
    RUNNING: 16,
    RESUME_TO_RUNNING: 18,
    PAUSED: 32,
    PAUSED_FROM_STARTING: 36,
    PAUSING_FROM_STARTING: 38,
    PAUSED_FROM_RUNNING: 48,
    PAUSING_FROM_RUNNING: 50,
    COMPLETED: 64,
    COMPLETING: 66,
    PREPARING: 68,
    ABORTED: 128,
    ABORTING: 130,
    ALARM: 256,
    FATAL: 512,
    TOOLHEAD_CHANGE: 1024,
  },
  statusColor: {
    '-17': 'grey',
    '-10': 'grey',
    '-2': 'grey',
    '-1': 'orange',
    0: 'grey',
    1: 'grey',
    4: 'blue',
    6: 'blue',
    16: 'blue',
    18: 'blue',
    32: 'orange',
    36: 'orange',
    38: 'orange',
    48: 'orange',
    50: 'orange',
    64: 'green',
    66: 'blue',
    128: 'red',
    256: 'red',
    512: 'red',
    1024: 'orange',
  },
};
