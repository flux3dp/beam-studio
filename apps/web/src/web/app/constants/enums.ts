export enum OptionValues {
  TRUE = 'TRUE',
  FALSE = 'FALSE',
}

export default {
  OptionValues,
};
