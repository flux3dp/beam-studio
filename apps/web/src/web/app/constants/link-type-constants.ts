export const LINKTYPE_CORNER = 0;
export const LINKTYPE_SMOOTH = 1; // same direction, different dist
export const LINKTYPE_SYMMETRIC = 2; // same direction, same dist
export type NodeLinkType = typeof LINKTYPE_CORNER |
                           typeof LINKTYPE_SMOOTH |
                           typeof LINKTYPE_SYMMETRIC;
