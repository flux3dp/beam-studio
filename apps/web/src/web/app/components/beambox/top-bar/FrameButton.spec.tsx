import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { CanvasContext } from 'app/contexts/CanvasContext';
import { CanvasMode } from 'app/constants/canvasMode';

import FrameButton from './FrameButton';

const mockShowFramingModal = jest.fn();
jest.mock('app/components/dialogs/FramingModal', () => ({
  showFramingModal: (...args) => mockShowFramingModal(...args),
}));

jest.mock('helpers/useI18n', () => () => ({
  topbar: {
    frame_task: 'frame_task',
  },
}));

jest.mock('app/contexts/CanvasContext', () => ({
  CanvasContext: React.createContext({ mode: 1 }),
  CanvasMode: {
    Draw: 1,
    Preview: 2,
    PathPreview: 3,
  },
}));

const mockGetSelectedDevice = jest.fn();
jest.mock('app/views/beambox/TopBar/contexts/TopBarController', () => ({
  getSelectedDevice: () => mockGetSelectedDevice(),
}));

const mockOn = jest.fn();
jest.mock('helpers/shortcuts', () => ({
  on: (...args) => mockOn(...args),
}));

describe('test FrameButton', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should render correctly', async () => {
    const { container } = render(<FrameButton />);
    expect(container).toMatchSnapshot();
    fireEvent.click(container.querySelector('div[class*="button"]'));
    expect(mockShowFramingModal).toBeCalledTimes(1);
  });

  test('should render correctly with previewing mode', () => {
    const { container } = render(
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      <CanvasContext.Provider value={{ mode: CanvasMode.Preview } as any}>
        <FrameButton />
      </CanvasContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  test('shortcut should work', () => {
    render(<FrameButton />);
    expect(mockOn).toBeCalledTimes(1);
    expect(mockOn).toBeCalledWith(['F1'], expect.any(Function));
    const handler = mockOn.mock.calls[0][1];
    expect(mockShowFramingModal).not.toBeCalled();
    mockGetSelectedDevice.mockReturnValue({ model: 'fpm1' });
    handler();
    expect(mockShowFramingModal).toBeCalledTimes(1);
  });

  // TODO: move this to helpers/device/framing.ts
  // test('no element', async () => {
  //   mockGetDevice.mockResolvedValue({ device: { model: 'ado1', version: '4.1.7' } });
  //   mockGetWidth.mockReturnValue(4300);
  //   mockGetHeight.mockReturnValue(3000);
  //   mockGetExpansion.mockReturnValue([0, 0]);
  //   mockGetVisibleElementsAndBBoxes.mockReturnValue([]);
  //   const { container } = render(<FrameButton />);
  //   await act(async () => fireEvent.click(container.querySelector('div[class*="button"]')));
  //   await act(async () => jest.runAllTimers());
  //   expect(mockOpenMessage).toBeCalledTimes(1);
  //   expect(mockOpenMessage).toBeCalledWith({
  //     key: 'no-element-to-frame',
  //     level: 'info',
  //     content: 'Please add objects first',
  //     duration: 3,
  //   });
  // });

  // test('ador low laser', async () => {
  //   const { container } = render(<FrameButton />);
  //   mockGetWidth.mockReturnValue(4300);
  //   mockGetHeight.mockReturnValue(3000);
  //   mockGetExpansion.mockReturnValue([0, 0]);
  //   mockGetDevice.mockResolvedValue({ device: { model: 'ado1', version: '4.1.7' } });
  //   mockGetDeviceDetailInfo.mockResolvedValue({ head_type: 1 });
  //   mockRead.mockReturnValueOnce(null).mockReturnValueOnce(3).mockReturnValueOnce(0);
  //   mockGetDoorOpen.mockResolvedValue({ value: '1', cmd: 'play get_door_open', status: 'ok' });
  //   await act(async () => fireEvent.click(container.querySelector('div[class*="button"]')));
  //   await act(async () => jest.runAllTimers());
  //   expect(mockPopById).toBeCalledTimes(1);
  //   expect(mockGetAllLayers).toBeCalledTimes(1);
  //   expect(mockGetDevice).toBeCalledTimes(1);
  //   expect(mockCheckDeviceStatus).toBeCalledTimes(1);
  //   expect(mockOpenNonstopProgress).toBeCalledTimes(1);
  //   expect(mockGetDeviceDetailInfo).toBeCalledTimes(1);
  //   expect(mockRead).toBeCalledTimes(4);
  //   expect(mockRead).toHaveBeenNthCalledWith(1, 'module-offsets');
  //   expect(mockRead).toHaveBeenNthCalledWith(2, 'low_power');
  //   expect(mockRead).toHaveBeenNthCalledWith(3, 'rotary_mode');
  //   expect(mockRead).toHaveBeenNthCalledWith(4, 'enable-job-origin');
  //   expect(mockGetDoorOpen).toBeCalledTimes(1);
  //   expect(mockOpenMessage).toBeCalledTimes(1);
  //   expect(mockOpenMessage).toBeCalledWith({
  //     key: 'low-laser-warning',
  //     level: 'info',
  //     content: 'Please close the door cover to enable low laser for running frame.',
  //   });
  //   expect(mockUpdate).toBeCalledTimes(6);
  //   expect(mockEnterRawMode).toBeCalledTimes(1);
  //   expect(mockRawSetRotary).toBeCalledTimes(1);
  //   expect(mockRawHomeZ).not.toBeCalled();
  //   expect(mockRawHome).toBeCalledTimes(1);
  //   expect(mockRawStartLineCheckMode).toBeCalledTimes(1);
  //   expect(mockRawSetFan).toBeCalledTimes(1);
  //   expect(mockRawSetAirPump).toBeCalledTimes(1);
  //   expect(mockRawMoveZRelToLastHome).not.toBeCalled();
  //   expect(mockRawMove).toBeCalledTimes(5);
  //   expect(mockRawMove).toHaveBeenNthCalledWith(1, { x: 0, y: 1, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(2, { x: 2, y: 1, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(3, { x: 2, y: 4, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(4, { x: 0, y: 4, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(5, { x: 0, y: 1, f: 6000 });
  //   expect(mockRawEndLineCheckMode).toBeCalledTimes(1);
  //   expect(mockRawSetLaser).toBeCalledTimes(2);
  //   expect(mockRawSetLaser).toHaveBeenNthCalledWith(1, { on: true, s: 30 });
  //   expect(mockRawSetLaser).toHaveBeenNthCalledWith(2, { on: false, s: 0 });
  //   expect(mockRawSet24V).toBeCalledTimes(2);
  //   expect(mockRawSet24V).toHaveBeenNthCalledWith(1, true);
  //   expect(mockRawSet24V).toHaveBeenNthCalledWith(2, false);
  //   expect(mockRawLooseMotor).toBeCalledTimes(1);
  //   expect(mockEndRawMode).toBeCalledTimes(1);
  //   expect(mockKick).toBeCalledTimes(1);
  //   expect(mockRawSetWaterPump).not.toBeCalled();
  // });

  // test('ador low laser with rotary and mirror', async () => {
  //   const { container } = render(<FrameButton />);
  //   mockGetWidth.mockReturnValue(4300);
  //   mockGetHeight.mockReturnValue(3000);
  //   mockGetExpansion.mockReturnValue([0, 0]);
  //   mockGetDevice.mockResolvedValue({ device: { model: 'ado1', version: '4.1.7' } });
  //   mockGetDeviceDetailInfo.mockResolvedValue({ head_type: 1 });
  //   mockRead.mockReturnValueOnce(null).mockReturnValueOnce(3).mockReturnValueOnce(1);
  //   mockGetDoorOpen.mockResolvedValue({ value: '1', cmd: 'play get_door_open', status: 'ok' });
  //   mockGetPosition.mockReturnValue(10);
  //   await act(async () => fireEvent.click(container.querySelector('div[class*="button"]')));
  //   await act(async () => jest.runAllTimers());
  //   expect(mockPopById).toBeCalledTimes(1);
  //   expect(mockGetAllLayers).toBeCalledTimes(1);
  //   expect(mockGetDevice).toBeCalledTimes(1);
  //   expect(mockCheckDeviceStatus).toBeCalledTimes(1);
  //   expect(mockOpenNonstopProgress).toBeCalledTimes(1);
  //   expect(mockGetDeviceDetailInfo).toBeCalledTimes(1);
  //   expect(mockRead).toBeCalledTimes(5);
  //   expect(mockRead).toHaveBeenNthCalledWith(1, 'module-offsets');
  //   expect(mockRead).toHaveBeenNthCalledWith(2, 'low_power');
  //   expect(mockRead).toHaveBeenNthCalledWith(3, 'rotary_mode');
  //   expect(mockRead).toHaveBeenNthCalledWith(4, 'rotary-mirror');
  //   expect(mockRead).toHaveBeenNthCalledWith(5, 'enable-job-origin');
  //   expect(mockGetDoorOpen).toBeCalledTimes(1);
  //   expect(mockOpenMessage).toBeCalledTimes(1);
  //   expect(mockOpenMessage).toBeCalledWith({
  //     key: 'low-laser-warning',
  //     level: 'info',
  //     content: 'Please close the door cover to enable low laser for running frame.',
  //   });
  //   expect(mockUpdate).toBeCalledTimes(6);
  //   expect(mockEnterRawMode).toBeCalledTimes(1);
  //   expect(mockRawHomeZ).toBeCalledTimes(1);
  //   expect(mockRawSetRotary).toBeCalledTimes(3);
  //   expect(mockRawSetRotary).toHaveBeenNthCalledWith(1, false);
  //   expect(mockRawSetRotary).toHaveBeenNthCalledWith(2, true);
  //   expect(mockRawSetRotary).toHaveBeenNthCalledWith(3, false);
  //   expect(mockRawHome).toBeCalledTimes(1);
  //   expect(mockRawStartLineCheckMode).toBeCalledTimes(1);
  //   expect(mockRawSetFan).toBeCalledTimes(1);
  //   expect(mockRawSetAirPump).toBeCalledTimes(1);
  //   expect(mockRawMoveZRelToLastHome).toBeCalledTimes(1);
  //   expect(mockRawMoveZRelToLastHome).toHaveBeenNthCalledWith(1, 0);
  //   expect(mockRawMove).toBeCalledTimes(8);
  //   expect(mockRawMove).toHaveBeenNthCalledWith(1, { x: 0, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(2, { y: 10, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(3, { x: 0, a: 19, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(4, { x: 2, a: 19, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(5, { x: 2, a: 16, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(6, { x: 0, a: 16, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(7, { x: 0, a: 19, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(8, { a: 10, f: 6000 });
  //   expect(mockRawEndLineCheckMode).toBeCalledTimes(1);
  //   expect(mockRawSetLaser).toBeCalledTimes(3);
  //   expect(mockRawSetLaser).toHaveBeenNthCalledWith(1, { on: true, s: 30 });
  //   expect(mockRawSetLaser).toHaveBeenNthCalledWith(2, { on: false, s: 0 });
  //   expect(mockRawSetLaser).toHaveBeenNthCalledWith(3, { on: false, s: 0 });
  //   expect(mockRawSet24V).toBeCalledTimes(2);
  //   expect(mockRawSet24V).toHaveBeenNthCalledWith(1, true);
  //   expect(mockRawSet24V).toHaveBeenNthCalledWith(2, false);
  //   expect(mockRawLooseMotor).toBeCalledTimes(1);
  //   expect(mockEndRawMode).toBeCalledTimes(1);
  //   expect(mockKick).toBeCalledTimes(1);
  //   expect(mockRawSetWaterPump).not.toBeCalled();
  // });

  // test('ador with job origin', async () => {
  //   const { container } = render(<FrameButton />);
  //   mockGetWidth.mockReturnValue(4300);
  //   mockGetHeight.mockReturnValue(3000);
  //   mockGetExpansion.mockReturnValue([0, 0]);
  //   mockGetDevice.mockResolvedValue({ device: { model: 'ado1', version: '5.3.5' } });
  //   mockGetDeviceDetailInfo.mockResolvedValue({ head_type: 1 });
  //   mockRead.mockImplementation((key) => {
  //     if (key === 'low_power') return 3;
  //     if (key === 'rotary_mode') return 0;
  //     if (key === 'enable-job-origin') return 1;
  //     return null;
  //   });
  //   mockGetDoorOpen.mockResolvedValue({ value: '1', cmd: 'play get_door_open', status: 'ok' });
  //   mockGetPosition.mockReturnValue(10);
  //   mockGetJobOrigin.mockReturnValue({ x: 100, y: 100 });
  //   await act(async () => fireEvent.click(container.querySelector('div[class*="button"]')));
  //   await act(async () => jest.runAllTimers());
  //   expect(mockPopById).toBeCalledTimes(1);
  //   expect(mockGetAllLayers).toBeCalledTimes(1);
  //   expect(mockGetDevice).toBeCalledTimes(1);
  //   expect(mockCheckDeviceStatus).toBeCalledTimes(1);
  //   expect(mockOpenNonstopProgress).toBeCalledTimes(1);
  //   expect(mockGetDeviceDetailInfo).toBeCalledTimes(1);
  //   expect(mockRead).toBeCalledTimes(4);
  //   expect(mockRead).toHaveBeenNthCalledWith(1, 'module-offsets');
  //   expect(mockRead).toHaveBeenNthCalledWith(2, 'low_power');
  //   expect(mockRead).toHaveBeenNthCalledWith(3, 'rotary_mode');
  //   expect(mockRead).toHaveBeenNthCalledWith(4, 'enable-job-origin');
  //   expect(mockGetDoorOpen).toBeCalledTimes(1);
  //   expect(mockOpenMessage).toBeCalledTimes(1);
  //   expect(mockOpenMessage).toBeCalledWith({
  //     key: 'low-laser-warning',
  //     level: 'info',
  //     content: 'Please close the door cover to enable low laser for running frame.',
  //   });
  //   expect(mockUpdate).toBeCalledTimes(6);
  //   expect(mockEnterRawMode).toBeCalledTimes(1);
  //   expect(mockRawHomeZ).not.toBeCalled();
  //   expect(mockRawSetRotary).toBeCalledTimes(1);
  //   expect(mockRawSetRotary).toHaveBeenNthCalledWith(1, false);
  //   expect(mockRawHome).not.toBeCalled();
  //   expect(mockRawUnlock).toBeCalledTimes(1);
  //   expect(mockRawSetOrigin).toBeCalledTimes(1);
  //   expect(mockRawStartLineCheckMode).toBeCalledTimes(1);
  //   expect(mockRawSetFan).toBeCalledTimes(1);
  //   expect(mockRawSetAirPump).toBeCalledTimes(1);
  //   expect(mockRawMoveZRelToLastHome).not.toBeCalled();
  //   expect(mockRawMove).toBeCalledTimes(6);
  //   expect(mockRawMove).toHaveBeenNthCalledWith(1, { x: -100, y: -99, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(2, { x: -98, y: -99, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(3, { x: -98, y: -96, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(4, { x: -100, y: -96, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(5, { x: -100, y: -99, f: 6000 });
  //   expect(mockRawMove).toHaveBeenNthCalledWith(6, { x: 0, y: 0, f: 6000 });
  //   expect(mockRawEndLineCheckMode).toBeCalledTimes(1);
  //   expect(mockRawSetLaser).toBeCalledTimes(2);
  //   expect(mockRawSetLaser).toHaveBeenNthCalledWith(1, { on: true, s: 30 });
  //   expect(mockRawSetLaser).toHaveBeenNthCalledWith(2, { on: false, s: 0 });
  //   expect(mockRawSet24V).toBeCalledTimes(2);
  //   expect(mockRawSet24V).toHaveBeenNthCalledWith(1, true);
  //   expect(mockRawSet24V).toHaveBeenNthCalledWith(2, false);
  //   expect(mockRawLooseMotor).toBeCalledTimes(1);
  //   expect(mockEndRawMode).toBeCalledTimes(1);
  //   expect(mockKick).toBeCalledTimes(1);
  //   expect(mockRawSetWaterPump).not.toBeCalled();
  // });

  // test('framing with promark & swiftray', async () => {
  //   const mockConsoleLog = jest.spyOn(console, 'log').mockImplementation();
  //   const { container } = render(<FrameButton />);
  //   mockGetDevice.mockResolvedValue({ device: { model: 'fpm1', version: '12.34.56' } });
  //   await act(async () => fireEvent.click(container.querySelector('div[class*="button"]')));
  //   expect(mockGetDevice).toBeCalledTimes(1);
  //   expect(mockCheckDeviceStatus).toBeCalledTimes(1);
  //   expect(mockFetchFraming).toBeCalledTimes(1);
  //   expect(mockStartFraming).toBeCalledTimes(1)
  //   expect(mockStopFraming).not.toBeCalled();
  //   expect(mockConsoleLog).toBeCalledTimes(2);
  //   expect(mockConsoleLog).toHaveBeenNthCalledWith(1, 'start framing upload');
  //   expect(mockConsoleLog).toHaveBeenNthCalledWith(2, 'start framing');
  //   await act(async () => fireEvent.click(container.querySelector('div[class*="button"]')));
  //   expect(mockStopFraming).toBeCalledTimes(1);
  //   expect(mockStartFraming).toBeCalledTimes(1);
  //   // make sure original framing is not called
  //   expect(mockRead).not.toBeCalled();
  //   expect(mockRawHome).not.toBeCalled();
  // });
});
