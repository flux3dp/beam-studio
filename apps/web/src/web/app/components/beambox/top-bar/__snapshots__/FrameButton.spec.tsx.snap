// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test FrameButton should render correctly 1`] = `
<div>
  <div
    class="button"
    title="frame_task"
  >
    <svgr-url />
  </div>
</div>
`;

exports[`test FrameButton should render correctly with previewing mode 1`] = `
<div>
  <div
    class="button disabled"
    title="frame_task"
  >
    <svgr-url />
  </div>
</div>
`;
